src/Compilers/Core/Portable/MetadataReader/MetadataTypeName.cs

```diff
@@ -195,6 +229,34 @@ public string TypeName
             }
         }
 
+        public ReadOnlyMemory<char> NamespaceNameMemory
+        {
+            get
+            {
+                if (_namespaceNameMemory.Equals(default))
```

Comment:
ROM.Empty is exactly the same as `default`.   However, ROM.Empty is not equal to `"".AsMemory()`: https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK4wD4AEBMBGAWACgsBmAAl3IGFyBvY8pynHex5zrPATgAoASgFMIAEwDyMADYBPALJCwAe2QyAPAGMAFhGQA+AHQBRMAAcEM4wEdUEKQGc+AIicGAgvYXLVfAJT+Abg4mAF9iEKA=

So this allows us to know if hte ROM has been initialized or not.  And once initialized to anything (including the ROM for an empty string), this will be false.

Note: this is why i'm *not* using ROM.Empty in the code as i think there is too much confusino with if htat means "default" (which it does) or if it means "equivalent to an empty string" (which it does not).